### DISCUSSION TITLE
git, gradle and all other automated command tools don't work

### DISCUSSION THREAD

**Discussion by 1337-haxxor** [coder/code-server#3722] [Q&A]
I don't know if I'm messing something up but when ever I try to use gitlens or gradle in code-server it fails due to having insufficient permissions. So git checkout in the terminal does not work but if i do sudo git checkout it does. I am using the docker version so i think it might be that docker doesn't have correct permissions to use those commands. if anybody has a fix and can tell me that would be amazing.

**Comment by jsjoeio** [comment]
Hmm...I wish I knew any easy way to check Docker permissions to figure out what might be wrong. Have you tried looking through previous discussions? There might be something in here that could help

**Reply by 1337-haxxor** [reply]
I was looking at that and I found something about how his git login was messed up but I'll go back to see if it's ￼applicable

**Reply by 1337-haxxor** [reply]
Yup there issue was related to ssh keys.

**Reply by jsjoeio** [reply]
Awesome! Glad you figured it out

**Reply by 1337-haxxor** [reply]
I'm saying that there issue is unrelated to mind. I'll keep digging. I prob just need to mess with perms

**Reply by jsjoeio** [reply]
Oh sorry, I misunderstood. Okay, let us know what you find. In the meantime, hopefully someone from the community might pop in with a suggestion

**Reply by 1337-haxxor** [reply]
So I found the issue. It's nothing to do with the container. Just that for some reason the commands can't run without sudo permission as if the file path is protected. Could it be that it's owned by a different user? In any case. How would I fix it. Im very new to linux and docker as a whole so sorry for being dumb.

**Reply by jsjoeio** [reply]
> so sorry for being dumb.

No need to apologize — a lot of this is new to me too!

Nice debugging! That could be it. Here's a [reply](https://github.com/cdr/code-server/discussions/3055#discussioncomment-577942) to another discussion about changing the user.

And here is a [reply](https://github.com/cdr/code-server/discussions/3623#discussioncomment-880725) with a couple links to check out. Let me know if either of those help!

**Reply by 1337-haxxor** [reply]
So I have gotten a lot of this working but I still can't do git clone as it says remote repository not found. This is on a private repo which isn't owned by me but on my other computers it works fine.

**Reply by 1337-haxxor** [reply]
Also gradle is working now

**Reply by jsjoeio** [reply]
Hmm...I've had that happen to me before. Are you signed in with GitHub?

**Reply by 1337-haxxor** [reply]
Yeah. Everything else works. My own private repos work. When I check the login then it shows my info

**Reply by jsjoeio** [reply]
Very strange...so to reproduce, we would need to try to clone a repo that is private but that we have read/write access to?

Do you mind creating a basic private repo and inviting me to it? Then I can try to reproduce

**Reply by 1337-haxxor** [reply]
I can't right now but I'll later tonight

**Reply by jsjoeio** [reply]
No worries! I'll here all year

**Reply by 1337-haxxor** [reply]
ok i sent the invite to the repo

**Reply by jsjoeio** [reply]
Thanks for doing that! I was able to clone it after signing in:

## Video

https://user-images.githubusercontent.com/3806031/125501533-4aa18fdb-2802-4907-b973-0ef202bbd726.mov

**Reply by 1337-haxxor** [reply]
Hmm. I'll poke around and see if there's any login issues I didn't see

**Reply by jsjoeio** [reply]
Sounds like a plan!
